### ISSUE TITLE
Measure number regions that don't start at 1, and drawMeasureNumbers option

### ISSUE THREAD

**Issue by finneganwl** [opensheetmusicdisplay/opensheetmusicdisplay#541]
I have run into 2 issues with measure numbers.

**Issue 1**
I tried to load a musicXml file where the first measure number is not 1, e.g. 

`<part id="P1">
    <measure number="100" ...`

and the measure numbers displayed still assume that the first measure is numbered 1. The rendering is ignoring the `number` property.

**Issue 2**
When loading a small score of only 1-2 measures, measure numbers are not displayed at all. It looks like by default, every other measure number is displayed starting with the 3rd measure. Should there exist a drawMeasureNumbers option in IOSMDOptions or another way to get finer grained control over how often to display measure numbers?

**Comment by bneumann** [comment]
Thanks for submitting this. I think I stumbled upon this when implementing the drawing method.
My proposal is to use more options as you say. Something like ```UseAutomaticMeasureNumbers``` (default: true) and ```DrawEveryNthMeasureNumber``` (default: 5).
We need the earlier because we now have the possibility to draw subsets of pieces where you don't want the old measure numbers to show.

**Comment by matt-uib** [comment]
@finneganwl yes currently we show the number on every second measure within every system, starting at each system start measure (without numbering the first measure of the first system) - if i'm correct.
Yes, makes total sense to have options for that for being able to customize this.
As this is rather a "nice to have" feature from what currently is pending, we would be very happy if someone of the community could take this over.
Would you like to maybe contribute on that?

**Comment by sschmidTU** [comment]
FYI: there are `drawMeasureNumbers` and `measureNumberInterval` options now in OSMDOptions.
But still, further `UseAutomaticMeasureNumbers` or something like `useXMLNumbers` options would be nice to have.

**Comment by sschmidTU** [comment]
Scores with measure numbers not starting at 1 should print the measure numbers given in XML now:
![image](https://user-images.githubusercontent.com/33069673/92279618-f9975a00-eef7-11ea-88fa-619de4d47141.png)
[measure_numbers_xml_starting_at_3_with_multirest.musicxml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/5176697/measure_numbers_xml_starting_at_3_with_multirest.musicxml.zip)

This is configured by `osmd.EngravingRules.UseXMLMeasureNumbers`, which defaults to true.

You can have basically arbitrary measure numbers as given in the XML now, even nonlinear and repeating:
![image](https://user-images.githubusercontent.com/33069673/92280656-2c425200-eefa-11ea-8eb7-82d7669ea2ff.png)
[measure_numbers_xml_nonlinear_order.musicxml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/5176736/measure_numbers_xml_nonlinear_order.musicxml.zip)
(set `osmd.EngravingRules.MeasureNumberLabelOffset = 1` to show the measure number for every measure)

This will cause `SourceMeasure.getPrintedMeasureNumber()` to return what was given in XML,
while `SourceMeasure.MeasureNumber` is unchanged and always starting at 1.
It makes sense to have one variable for measure numbers starting at 1 (or 0) representing the logical measure number, that is, "this measure is the n-th printed on screen", for array indexes etc. So the first measure printed on screen still always has `MeasureNumber` 1, while `MeasureNumberXML` and `MeasureNumberPrinted` (set in `getPrintedMeasureNumber()`) can be different.
